sha2/src/sha512/x86.rs

```diff
@@ -11,12 +11,13 @@ use core::arch::x86_64::*;
 
 use crate::consts::K64;
 
+cpufeatures::new!(avx_cpuid, "avx");
 cpufeatures::new!(avx2_cpuid, "avx2");
 
 pub fn compress(state: &mut [u64; 8], blocks: &[[u8; 128]]) {
     // TODO: Replace with https://github.com/rust-lang/rfcs/pull/2725
     // after stabilization
-    if avx2_cpuid::get() {
+    if avx_cpuid::get() && avx2_cpuid::get() {
```

Comment:
`cpufeatures` support detection of several features simultaneously, s you can simply change the `avx2_cpuid` creation macro to:
```rust
// We need to check both availability of both AVX and AVX2, see:
// https://github.com/RustCrypto/hashes/pull/386
cpufeatures::new!(avx2_cpuid, "avx", "avx2");
```
Without introducing `avx_cpuid`.

Reply:
I will update this.